L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.h

```diff
@@ -0,0 +1,450 @@
+    else if (alctRoll.second == -99 and !(padRoll >= alctRoll.first)) continue; // bottom of the chamber
+    else if ((alctRoll.first != -99 and alctRoll.second != -99) and // center
+	     (alctRoll.first > padRoll or padRoll > alctRoll.second)) continue;
+    result.push_back(p);
+  }
+  return result;
+}
+
+template <class S>
+matches<S> CSCGEMMotherboard::matchingPads(const CSCCLCTDigi& clct, enum CSCPart part)
+{
+  matches<S> result;
+  if (not clct.isValid()) return result;
+  
+  int deltaBX;
+  // get the generic LUT  
+  boost::variant<GEMPadDigiIdsBX, GEMCoPadDigiIdsBX> variant_lut;
+  if (std::is_same<S, GEMPadDigi>::value) {
+    variant_lut = pads_;
+    deltaBX = maxDeltaBXPad_;
+  }
+  else{
+    variant_lut = coPads_;
+    deltaBX = maxDeltaBXCoPad_;
+  }
+  const auto& actual_lut = boost::get<std::map<int, matches<S> > >(variant_lut);
+  
+  const auto& mymap = (getLUT()->get_csc_hs_to_gem_pad(par, part));
+  const int lowPad(mymap[clct.getKeyStrip()].first);
+  const int highPad(mymap[clct.getKeyStrip()].second);
+  for (const auto& p: actual_lut[clct.getBX()]){
+    auto padRoll(getAvePad(p.second));
+    int pad_bx = getBX(p.second)+lct_central_bx;
+    if (std::abs(clct.getBX()-pad_bx)>deltaBX) continue;
+    if (std::abs(lowPad - padRoll) <= maxDeltaPadL1_ or std::abs(padRoll - highPad) <= maxDeltaPadL1_){
+      result.push_back(p);
+    }
+  }
+  return result;
+}
+
+template <class S, class T>
+matches<T> CSCGEMMotherboard::matchingPads(const S& d1, const S& d2, enum CSCPart part)
+{
+  matches<T> result, p1, p2;
+  p1 = matchingPads<T>(d1, part);
+  p2 = matchingPads<T>(d2, part);
+  result.reserve(p1.size() + p2.size());
+  result.insert(std::end(result), std::begin(p1), std::end(p1));
+  result.insert(std::end(result), std::begin(p2), std::end(p2));
+  return result;
+}
+
+template <class T>
+matches<T> CSCGEMMotherboard::matchingPads(const CSCCLCTDigi& clct1, const CSCALCTDigi& alct1, enum CSCPart part)
+{
+  matches<T> padsClct(matchingPads<T>(clct1, part));
+  matches<T> padsAlct(matchingPads<T>(alct1, part));
+  return intersection(padsClct, padsAlct);
+}
+
+template <class T>
+matches<T> CSCGEMMotherboard::matchingPads(const CSCCLCTDigi& clct1, const CSCCLCTDigi& clct2,
+					   const CSCALCTDigi& alct1, const CSCALCTDigi& alct2,
+					   enum CSCPart part)
+{
+  matches<T> padsClct(matchingPads<CSCCLCTDigi,T>(clct1, clct2, part));
+  matches<T> padsAlct(matchingPads<CSCALCTDigi,T>(alct1, alct2, part));
+  return intersection(padsClct, padsAlct);
+}
+
+template <class S>  
+void CSCGEMMotherboard::printGEMTriggerPads(int bx_start, int bx_stop, enum CSCPart part)
+{
+  bool iscopad = false;
+  // get the generic LUT  
+  boost::variant<GEMPadDigiIdsBX, GEMCoPadDigiIdsBX> variant_lut;
+  if (std::is_same<S, GEMPadDigi>::value) variant_lut = pads_;
+  else {
+    variant_lut = coPads_;
+    iscopad = true;
+  }
+  const auto& actual_lut = boost::get<std::map<int, matches<S> > >(variant_lut);
```

Comment:
please put these lines into a helper function (rather complicated and repeated above)